File: arcade/python/arcade-intro/09_Dark Wilderness/039_knapsackLight.py
def knapsackLight(value1, weight1, value2, weight2, maxW):
        '''
    You found two items in a treasure chest! The first item weighs weight1 and is worth value1,
    and the second item weighs weight2 and is worth value2. What is the total maximum value of 
    the items you can take with you, assuming that your max weight capacity is maxW and you 
    can't come back for the items later?
    '''
    # Take both items
    if weight1 + weight2 <= maxW:
        return value1 + value2    
    
    # If we can choose from the two, chose the one with that values the most
    if weight1 <= maxW and weight2 <= maxW:
        return max(value1, value2)
    
    # If we can't decide choose the one we can carry
    if weight1 <= maxW:
        return value1
    if weight2 <= maxW:
        return value2
    
    # WE can't carry any item
    return 0

File: arcade/go/arcade-intro/09_Dark Wilderness/039_knapsackLight.go
func knapsackLight(value1 int, weight1 int, value2 int, weight2 int, maxW int) int {
	// if we can grab both, sum the values
	if weight1+weight2 <= maxW {
		return value1 + value2
	}

	// if we can choose, return the one with the highest value
	if weight1 <= maxW && weight2 <= maxW {
		if value1 > value2 {
			return value1
		}
		return value2
	}

	// if we cannot choose, return the one we can carry
	if weight1 <= maxW {
		return value1
	}
	if weight2 <= maxW {
		return value2
	}

	// we cannot carry either, return 0
	return 0
}
